doc/stdenv/platform-notes.chapter.md

```diff
@@ -2,21 +2,22 @@
 
-- The Darwin `stdenv` uses clang instead of gcc. When referring to the compiler `$CC` or `cc` will work in both cases. Some builds hardcode gcc/g++ in their build scripts, that can usually be fixed with using something like `makeFlags = [ "CC=cc" ];` or by patching the build scripts.
+- Darwin uses clang by default instead of GCC. Packages that refer to `$CC` or `cc` should just work in most cases. Some packages may hardcode `gcc` or `g++`. You can usually fix that by setting `makeFlags = [ "CC=cc" "CXX=C++" ]`. If that does not work, you will have to patch the build scripts yourself to use the correct compiler for Darwin.
+- Darwin needs an SDK to build software. The SDK provides a default set of frameworks and libraries to build software, most of which are specific to Darwin. There are multiple versions of the SDK packages in nixpkgs, but one is included by default in the `stdenv`. Usually, you don’t have to change or pick a different SDK. When in doubt, use the default.
+- The SDK used by your build can be found using the `DEVELOPER_DIR` environment variable. There are also versions of this variable available when cross-compiling depending on the SDK’s role. The `SDKROOT` variable is also set with the path to the SDK’s libraries and frameworks. `SDKROOT` is always a sub-folder of `DEVELOPER_DIR`.
+- Darwin includes a platform-specific tool called `xcrun` to help builds locate binaries they need. A version of `xcrun` is part of the `stdenv` on Darwin. If your package invokes `xcrun` via an absolute path (such as `/usr/bin/xcrun`), you will need to patch the build scripts to use `xcrun` instead.
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "libfoo-1.2.3";
-    # ...
-    buildPhase = ''
-      $CC -o hello hello.c
-    '';
-  }
-  ```
+To reiterate: you usually don’t have to worry about this stuff. Start with writing your derivation as if everything is already set up for you (because in most cases it already is). If you run into issues or failures, continue reading below for how to deal with the most common issues you may encounter.
+
+### Darwin Issue Troubleshooting
 
-- On Darwin, libraries are linked using absolute paths, libraries are resolved by their `install_name` at link time. Sometimes packages won’t set this correctly causing the library lookups to fail at runtime. This can be fixed by adding extra linker flags or by running `install_name_tool -id` during the `fixupPhase`.
+#### Library install name issues
+
+Libraries on Darwin are usually linked with absolute paths. This is determiend by something called an “install name”, which is resolved at link time. Sometimes packages will not set this correctly, causing binaries linking to it not to find their libraries at runtime. This can be fixed by adding extra linker flags or by using `install_name_tool` to set it in `fixupPhase`.
```

Comment:
Let’s recommend `fixDarwinDylibNames` over manual patching; would simplify this whole section. I also think it could go further down; having to add a newer SDK or `xcbuild` is much more common than this.

Reply:
I can move it down, but this was in the previous documentation. I’d also prefer not to recommend that hook because it normally shouldn’t be needed, and if it’s in the documentation, it might proliferate.

Reply:
I added a mention of `fixDarwinDylibNames` to the documentation in the install name section, which I moved to the end of the troubleshooting section.